src/marqo/core/index_management/index_management.py

```diff
@@ -75,35 +47,66 @@ def __init__(self, vespa_client: VespaClient, zookeeper_client: Optional[Zookeep
                 the object can create/delete indexes, otherwise, it raises an InternalError during index operations.
         """
         self.vespa_client = vespa_client
-        self._zookeeper_client = zookeeper_client
-        self._zookeeper_deployment_lock: Optional[ZookeeperDistributedLock] = self._instantiate_deployment_lock()
+        self._zookeeper_deployment_lock = get_deployment_lock(zookeeper_client, deployment_lock_timeout) \
+            if zookeeper_client else None
         self._enable_index_operations = enable_index_operations
+        self._deployment_timeout_seconds = deployment_timeout_seconds
+        self._convergence_timeout_seconds = convergence_timeout_seconds
 
-    def _instantiate_deployment_lock(self) -> Optional[ZookeeperDistributedLock]:
-        """Instantiate a ZookeeperDistributedLock."""
-        if self._zookeeper_client is None:
-            return None
-        else:
-            return get_deployment_lock(self._zookeeper_client)
+    @staticmethod
```

Comment:
Does this need to be static? I tend to avoid one random static method just because it doesn't need any instance variables, to keep things consistent. I could be wrong though

Reply:
no, will change